ddtrace/profiling/_periodic.py

```diff
@@ -32,24 +33,23 @@ def __init__(self, interval, target, name=None, on_shutdown=None):
         self.quit = threading.Event()
         self.daemon = True
 
-    def start(self):
-        """Start the thread."""
-        super(PeriodicThread, self).start()
-        PERIODIC_THREAD_IDS.add(self.ident)
-
     def stop(self):
         """Stop the thread."""
         self.quit.set()
 
     def run(self):
         """Run the target function periodically."""
+        with id_lock:
+            PERIODIC_THREAD_IDS.add(self.ident)
```

Comment:
I tried to find a better hook in to add the id, but there really isn't any unless we override `bootstrap` and/or `_bootstrap`.

Reply:
I don't think you need a lock there, or I don't see why?

Reply:
`PERIODIC_THREAD_IDS` is global and shared amongst each periodic worker. So if we have two (which we do: 1 in the tracer, 1 in the profiler, I think) it's possible that they both modify the variable at the same time. AFAIK `set` operations are not threadsafe.

It's a little unnecessary because these periodic workers don't start and stop at the same time but it might save a headache or two down the line if we change our usage.

These were my thoughts here 🙂

Reply:
`set` operations are thread-safe — just like all builtins data structure.
You're right we'd need a lock if they wouldn't be though.

Reply:
ahhh :+1: I'll remove the lock